examples/ct/ct_handlers_test.go

```diff
@@ -972,11 +1135,23 @@ func leafProtosForCert(t *testing.T, km crypto.KeyManager, certs []*x509.Certifi
 		t.Fatalf("failed to serialize log entry: %v", err)
 	}
 
-	return []*trillian.LeafProto{&trillian.LeafProto{LeafHash: leafHash[:], LeafData: b.Bytes(), ExtraData: b2.Bytes()}}
+	return []*trillian.LeafProto{{LeafHash: leafHash[:], LeafData: b.Bytes(), ExtraData: b2.Bytes()}}
+}
+
+type dlMatcher struct {
+}
+
+func deadlineMatcher() gomock.Matcher {
+	return dlMatcher{}
 }
 
-func deadlineMatcher(other context.Context) bool {
-	deadlineTime, ok := other.Deadline()
+func (d dlMatcher) Matches(x interface{}) bool {
```

Comment:
I'll leave it for now on the "like-for-like(-ish)" argument, and we can sort it out in round two?